docs/derivation-rules.md

```diff
@@ -0,0 +1,71 @@
+This file shows derivation rules used in fialyzer.
+
+Our derivation rules are almost same as [the original paper](https://it.uu.se/research/group/hipe/papers/succ_types.pdf)'s one, but extended by remote call, local call, list, etc.
+
+```
+A : context (mapping of variable to type)
+τ : type
+α, β : type variable
+x : variable
+e : term
+fun f/a : local function (f: function name, a: arity)
+fun m:f/a : remote function (m: module name, f: function name, a: arity)
+```
```

Comment:
it is good if `p : pattern` is written.

Reply:
Indeed. Thanks!

Reply:
fixed https://github.com/dwango/fialyzer/pull/164/commits/5bb44eba7ebf07f37788d64d91ec002dcf52f645